File: core/src/main/java/dev/bee/kanjianki/core/SettingsLadderThresholdTextCopy.java
package dev.bee.kanjianki.core;

import java.util.Locale;

public final class SettingsLadderThresholdTextCopy {
    private SettingsLadderThresholdTextCopy() {
    }

    public static String ladderThresholdsTitle() {
        return "Ladder thresholds";
    }

    public static String ladderThresholdsBody() {
        return "Recognition rungs climb when a real FSRS-due pass schedules the next review beyond the day threshold. Learning-step repeats stay practice-only.";
    }

    public static String fsrsDaysToGoUpLabel() {
        return "FSRS days to go up";
    }

    public static String failsToGoDownLabel() {
        return "Fails to go down";
    }

    public static String useDefaultLadderThresholdsLabel() {
        return String.format(
                Locale.ROOT,
                "Use %d and %d",
                RecordsBase.DEFAULT_LADDER_PROMOTION_INTERVAL_DAYS,
                RecordsBase.DEFAULT_LADDER_DEMOTION_FAIL_STREAK
        );
    }

    public static String saveLadderThresholdsLabel() {
        return "Save ladder thresholds";
    }

    public static String ladderThresholdsSavedToast() {
        return "Ladder thresholds saved.";
    }
}

File: core/src/main/kotlin/dev/bee/kanjianki/core/SettingsLadderThresholdTextCopy.kt
package dev.bee.kanjianki.core

import java.util.Locale

object SettingsLadderThresholdTextCopy {
    private const val JAPANESE_LANGUAGE = "ja"

    @JvmStatic
    fun ladderThresholdsTitle(): String = localizedText("Ladder movement", "ラダー移動")

    @JvmStatic
    fun ladderThresholdsBody(): String = localizedText("Due reviews move cards. Repeats stay practice-only.", "期限レビューでカードが移動する。繰り返しは練習のみ。")

    @JvmStatic
    fun fsrsDaysToGoUpLabel(): String = localizedText("Days to move up", "上がるまでの日数")

    @JvmStatic
    fun failsToGoDownLabel(): String = localizedText("Fails to move down", "下がるまでの失敗数")

    @JvmStatic
    fun useDefaultLadderThresholdsLabel(): String = localizedText("Use default movement rules", "既定の移動ルールを使う")

    @JvmStatic
    fun saveLadderThresholdsLabel(): String = localizedText("Save rules", "ルールを保存")

    @JvmStatic
    fun ladderThresholdsSavedToast(): String = localizedText("Movement rules saved.", "移動ルールを保存しました。")

    private fun localizedText(english: String, japanese: String): String =
        if (isJapaneseLocale()) japanese else english

    private fun isJapaneseLocale(): Boolean = Locale.getDefault().language == JAPANESE_LANGUAGE
}
